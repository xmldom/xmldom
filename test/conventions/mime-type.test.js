'use strict'

const { MIME_TYPE } = require('../../lib/conventions')

describe('MIME_TYPE', () => {
	Object.keys(MIME_TYPE).forEach((key) => {
		if (key === 'isHTML') {
			describe('isHTML', () => {
				it("should return true for 'text/html'", () => {
					expect(MIME_TYPE.isHTML('text/html')).toBe(true)
				})
				it('should return true for MIME_TYPE.HTML', () => {
					expect(MIME_TYPE.isHTML(MIME_TYPE.HTML)).toBe(true)
				})
				it.each([
					undefined,
					null,
					0,
					1,
					false,
					true,
					'',
					MIME_TYPE.XML_XHTML_APPLICATION,
				])("should return false for '%s'", (value) => {
					expect(MIME_TYPE.isHTML(value)).toBe(false)
				})
			})
		} else {
			const value = MIME_TYPE[key]
			it(`should contain immutable ${key} with correct value`, () => {
				expect([key, value]).toMatchSnapshot()
				try {
					MIME_TYPE[key] = 'boo'
				} catch {}
				expect(MIME_TYPE[key]).toBe(value)
			})
		}
	})
})
