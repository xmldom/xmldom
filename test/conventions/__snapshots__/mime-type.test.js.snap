// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MIME_TYPE should contain immutable HTML with correct value 1`] = `
Array [
  "HTML",
  "text/html",
]
`;

exports[`MIME_TYPE should contain immutable XML_APPLICATION with correct value 1`] = `
Array [
  "XML_APPLICATION",
  "application/xml",
]
`;

exports[`MIME_TYPE should contain immutable XML_SVG_IMAGE with correct value 1`] = `
Array [
  "XML_SVG_IMAGE",
  "image/svg+xml",
]
`;

exports[`MIME_TYPE should contain immutable XML_TEXT with correct value 1`] = `
Array [
  "XML_TEXT",
  "text/xml",
]
`;

exports[`MIME_TYPE should contain immutable XML_XHTML_APPLICATION with correct value 1`] = `
Array [
  "XML_XHTML_APPLICATION",
  "application/xhtml+xml",
]
`;
